File: src/main/java/pers/lyks/spring/example/strategy/AddOperation.java
package pers.lyks.spring.example.strategy;

import org.springframework.stereotype.Component;

@Component
@CalculateHandlerType("add")
public class AddOperation implements CalculateStrategy {
    @Override
    public Number calculate(Number first, Number second) {
        return first.doubleValue() + second.doubleValue() ;
    }
}

File: src/main/kotlin/pers/lyks/spring/example/strategy/AddOperation.kt
package pers.lyks.spring.example.strategy

import org.springframework.stereotype.Component

@Component
@CalculateHandlerType("add")
class AddOperation : CalculateStrategy<Double, Double> {
    override fun calculate(first: Double?, second: Double?): Number {
        return first!! + second!!
    }
}
